unicorn/templates/default/unicorn.service.erb

```diff
@@ -0,0 +1,140 @@
+#!/usr/local/bin/ruby
+# unicorn restart script
+# source at https://github.com/sdtechdev/frt-opsworks-cookbooks/blob/update_unicorn_script/unicorn/templates/default/unicorn.service.erb
+# on May 27, 2021
+
+require 'etc'
+require 'digest/md5'
+require 'open3'
+require 'logger'
+
+ROOT_PATH="<%= @deploy[:deploy_to] %>"
+APP_NAME="<%= @application %>"
+PID_PATH="<%= @deploy[:deploy_to] %>/shared/pids/unicorn.pid"
+LOG_PATH="<%= @deploy[:deploy_to] %>/shared/log/unicorn.restart.log"
+
+def run_and_print_command(command)
+  log command
+  result = system(command)
+  _stdout, stderr, status = Open3.capture3(command)
+
+  if status.success?
+    log "success"
+  else
+    log "failure, exitcode #{status.exitstatus}\n#{stderr}"
+    exit(1)
+  end
+end
+
+def run_and_ignore_exitcode_and_print_command(command)
+  log command
+  _stdout, stderr, status = Open3.capture3(command)
+
+  if status.success?
+    log "success"
+  else
+    log "failure, exitcode #{status.exitstatus}\n#{stderr}"
+  end
+end
+
+def unicorn_running?
+  if File.exists?(PID_PATH) && (pid = File.read(PID_PATH).chomp) && system("ps aux | grep #{pid} | grep -v grep > /dev/null")
+    pid
+  else
+    false
+  end
+end
+
+def different_gemfile?
+  if File.exists?("#{ROOT_PATH}/current/Gemfile")
+    dir = Dir["#{ROOT_PATH}/releases/*"]
+    previous_release_path = dir.sort[dir.size-2]
+    if !previous_release_path.nil? && File.exists?("#{previous_release_path}/Gemfile")
+      return Digest::MD5.hexdigest(File.read("#{ROOT_PATH}/current/Gemfile")) != Digest::MD5.hexdigest(File.read("#{previous_release_path}/Gemfile"))
+    end
+  end
+  false
+end
+
+def start_unicorn
+  if File.exists?("#{ROOT_PATH}/current/Gemfile")
+    log "OpsWorks: Gemfile detected - running Unicorn with bundle exec"
+    run_and_ignore_exitcode_and_print_command "cd #{ROOT_PATH}/current && /usr/local/bin/bundle exec unicorn_rails --env <%= @deploy[:rails_env] %> --daemonize -c #{ROOT_PATH}/shared/config/unicorn.conf"
+  else
+    log "OpsWorks: no Gemfile detected - running plain Unicorn"
+    run_and_ignore_exitcode_and_print_command "cd #{ROOT_PATH}/current && unicorn_rails --env <%= @deploy[:rails_env] %> --daemonize -c #{ROOT_PATH}/shared/config/unicorn.conf"
+  end
+end
+
+def stop_unicorn
+  if unicorn_running?
+    if run_and_ignore_exitcode_and_print_command "kill -QUIT `cat #{PID_PATH}`"
+      `rm #{PID_PATH}`
+    end
+  else
+    log "You can't stop unicorn, because it's not running"
+  end
+end
+
+def restart_unicorn
+	if unicorn_running?
+		run_and_ignore_exitcode_and_print_command "kill -USR2 `cat #{PID_PATH}`"
+	else
+		start_unicorn
+	end
+end
```

Comment:
do we want additional logging for this command, because it causes issues?

Reply:
I added more logging to `unicorn_running?` check, so any low-level action will be logged, I suppose it will be enough